pkg/gui/gui_driver.go

```diff
@@ -59,6 +63,12 @@ func (self *GuiDriver) waitTillIdle() {
 	<-self.isIdleChan
 }
 
+func (self *GuiDriver) checkAllToastsAcknowledged() {
+	if t := self.NextToast(); t != nil {
+		self.Fail("Toast not acknowledged: " + *t)
+	}
+}
+
```

Comment:
The main reason I want to do this is for the case where a test creates several toasts; you forget to assert the first one, and then get a confusing failure message for the second one.

I'm checking this before every click and every key press; any other places that might need the check? Maybe at the very end of the test?

Reply:
I was about to say I'd rather not force test writers to have to always assert on toasts, but the only places we're currently using toasts are:
* copying to clipboard
* deleting a remote tag
* changing diff context size

And when it comes to error toasts we would always want an assertion on that. So I think it's fine.

If we're going to do this then yes we should also have an assertion at the end of the test, if only so that we don't create a perverse incentive to not assert on any toasts in the event that the test writer cares about one toast but not others in the same test.

Reply:
> I was about to say I'd rather not force test writers to have to always assert on toasts,

I was unsure too, but then I just tried it to see how many tests I would have to adapt, found that it wasn't many, and decided it's ok.

> If we're going to do this then yes we should also have an assertion at the end of the test, 

Any suggestion where the best place is to put that test?

Reply:
You can put it after `self.run(testDriver, keys)` in `pkg/integration/components/test.go:195`

Reply:
Done in ef77dd86c452. (I had to make `CheckAllToastsAcknowledged` public for this.)